File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
import sys

N= sys.stdin.readline().strip()

M = N.split('-')

answer = 0

x= sum(map(int,(M[0].split('+'))))
if N[0] == "-" :
    answer = answer-x
else:
    answer = answer+x

for x in M[1:] :

    x = sum(map(int, (x.split('+'))))
    answer = answer-x

print(answer)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.util.*;
import java.lang.*;
import java.io.*;

// The main method must be in a class named "Main".
class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br= new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        String basic = br.readLine();

        String [] oneline = basic.split("\\-");

        Long answer = 0L ;
        String[]  firstarr = oneline[0].split("\\+");
        for (int i = 0 ; i<firstarr.length;i++)
            {
                answer+= Long.parseLong(firstarr[i]);
            }

        for (int i =1; i<oneline.length;i++)
            {
                String [] twoarr = oneline[i].split("\\+");

                long smallanswer = 0;
                for (int j = 0 ; j<twoarr.length;j++)
                {
                    smallanswer+= Long.parseLong(twoarr[j]);
                }
                answer -= smallanswer;

            }

        bw.write(String.valueOf(answer));
        bw.close();

    }


}